batchflow/models/torch/base.py

```diff
@@ -710,11 +712,59 @@ def head(cls, inputs, target_shape, classes, **kwargs):
         """
         _ = target_shape, classes
         kwargs = cls.get_defaults('head', kwargs)
-        if kwargs.get('layout'):
+        if kwargs.get('layout') or kwargs.get('base_block'):
             return ConvBlock(inputs=inputs, **kwargs)
         return None
 
 
+    def get(self, name, from_start=False, wrap=True):
+        """ Get parts from model.
+
+        Parameters
+        ----------
+        name : str
+            Name of model part to retrieve.
+        from_start : bool
+            If True, then model is retrieved up to the part specified by `name`.
+            If False, only specified part is retrieved.
+        wrap : bool
+            Whether to wrap returned module into separate class that returns only the last output.
+        """
+        if self.model is None:
+            raise ValueError('Model is not created yet.')
+
+        order = self.full_config['order']
+
+        if name in order:
+            if from_start is False:
+                blocks = [(name, getattr(self.model, name) if hasattr(self.model, name) else None)]
+            if from_start is True:
+                index = order.index(name) + 1
+                blocks = [(item, getattr(self.model, item)) for item in order[:index] if hasattr(self.model, item)]
+
+        if name == 'encoder':
```

Comment:
That's a bunch of very strange hard-code which we should get rid of

Reply:
The original intent for this function is to exactly retrieve encoder and make it so it can be reused in other models by optionally wrapping it. Other functionality -- retrieving parts of the model  -- did come later. We can get rid of this function altogether, if we think that it is better to delegate these easy wrappings to the user